src/EditorFeatures/Core/GoToDefinition/AbstractGoToDefinitionService.cs

```diff
@@ -17,9 +18,9 @@ namespace Microsoft.CodeAnalysis.Editor.GoToDefinition
     // GoToDefinition
     internal abstract class AbstractGoToDefinitionService : IGoToDefinitionService
     {
-        private readonly IStreamingFindUsagesPresenter _streamingPresenter;
+        private readonly Lazy<IStreamingFindUsagesPresenter> _streamingPresenter;
```

Comment:
can we doc that this needs to be realized on hte UI thread only?

Reply:
It no longer needs to be constructed on the UI thread (since I moved the get service call), but in the lsp server case it doesn't need to be constructed at all => therefore lazily create it

Reply:
I mean: doc it in the code. :)